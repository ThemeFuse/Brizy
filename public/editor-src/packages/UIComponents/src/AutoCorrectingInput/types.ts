import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
