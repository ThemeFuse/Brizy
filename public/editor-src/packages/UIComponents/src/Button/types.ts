import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
