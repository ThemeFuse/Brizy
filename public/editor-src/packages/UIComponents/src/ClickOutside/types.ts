import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
