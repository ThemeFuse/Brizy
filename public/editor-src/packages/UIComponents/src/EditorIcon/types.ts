export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
