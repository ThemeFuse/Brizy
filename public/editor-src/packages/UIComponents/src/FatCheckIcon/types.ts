import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
