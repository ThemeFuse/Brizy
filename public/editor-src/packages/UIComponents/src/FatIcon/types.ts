import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
