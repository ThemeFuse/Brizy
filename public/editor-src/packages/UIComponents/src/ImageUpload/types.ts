import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onChange: () => void;
}
