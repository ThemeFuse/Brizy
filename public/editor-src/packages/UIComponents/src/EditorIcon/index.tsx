import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
