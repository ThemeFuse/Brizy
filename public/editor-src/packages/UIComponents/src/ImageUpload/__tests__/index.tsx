import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import { EditorIcon } from "../../EditorIcon";
import { IconsName } from "../../EditorIcon/types";
import { ImageUpload } from "../index";

describe("Test ImageUpload", () => {
  const onChange = jest.fn();
  const component = (
    <ImageUpload onChange={onChange}>
      <EditorIcon
        icon={IconsName.PlusCircle}
        className="brz-ed-control__gallery__item__icon"
      />
    </ImageUpload>
  );

  describe("Snapshots", () => {
    test("Image upload renders correctly", async () => {
      const { container } = render(component);

      await waitFor(() => {
        expect(container.firstChild).toBeInTheDocument();
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test("Check click event", async () => {
    const { container } = render(component);
    const element = container.firstElementChild;
    await waitFor(() => {
      expect(container.firstChild).toBeInTheDocument();
    });

    element && fireEvent.click(element);

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
