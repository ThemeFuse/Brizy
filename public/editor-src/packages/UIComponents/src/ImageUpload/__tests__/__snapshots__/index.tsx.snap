// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ImageUpload Snapshots Image upload renders correctly 1`] = `
<label
  class="brz-ed-control__imageUpload"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial] brz-ed-control__gallery__item__icon"
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <path
        d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
        fill="currentColor"
      />
    </g>
  </svg>
</label>
`;
